contracts/utils/math/Math.sol

```diff
@@ -335,6 +330,59 @@ library Math {
         }
     }
 
+    /**
+     * @dev Variant of {modExp} that supports inputs of arbitrary length.
+     */
+    function modExp(bytes memory b, bytes memory e, bytes memory m) internal view returns (bytes memory) {
+        (bool success, bytes memory result) = tryModExp(b, e, m);
+        if (!success) {
+            Panic.panic(Panic.DIVISION_BY_ZERO);
+        }
+        return result;
+    }
+
+    /**
+     * @dev Variant of {tryModExp} that supports inputs of arbitrary length.
+     */
+    function tryModExp(
+        bytes memory b,
+        bytes memory e,
+        bytes memory m
+    ) internal view returns (bool success, bytes memory result) {
+        if (_zeroBytes(m)) return (false, result);
+
+        uint256 mLen = m.length;
+
+        // Encode call args in result and move the free memory pointer
+        result = abi.encodePacked(b.length, e.length, mLen, b, e, m);
+
+        /// @solidity memory-safe-assembly
+        assembly {
+            // Write result on top of args to avoid allocating extra memory.
+            // | Offset    | Content      | Content (Hex)                                                      |
+            // |-----------|--------------|--------------------------------------------------------------------|
+            // | 0x00:0x1f | args length  | 0x<.......................................20+20+20+bLen+eLen+mLen> |
+            // | 0x20+mLen | result       | 0x<........................................................result> |
+            // | 0x..:0x.. | dirty bytes  | 0x<............................................20+20+20+bLen+eLen> |
```

Comment:
I'm not sure this table really helps. The last line in particular. Offsets are not specified, and the hex content is not accurate.

Reply:
Yeah, I started with a bigger table when using your example. I removed it.